types.go

```diff
@@ -493,3 +493,53 @@ type VSP struct {
 }
 
 /** end vspd-related types */
+
+/** begin agenda types */
+
+// Agenda contains information about a consensus deployment
+type Agenda struct {
+	AgendaID         string        `json:"agenda_id"`
+	Description      string        `json:"description"`
+	Mask             uint32        `json:"mask"`
+	Choices          []*VoteChoice `json:"choices"`
+	VotingPreference string        `json:"voting_preference"`
+	StartTime        int64         `json:"start_time"`
+	ExpireTime       int64         `json:"expire_time"`
+	Status           string        `json:"status"`
+}
+
+// VoteChoice holds information about the possible choices that
+// an agenda can vote with.
+type VoteChoice struct {
+	Id          string `json:"id"`
+	Description string `json:"description"`
+	Bits        uint32 `json:"bits"`
+	IsAbstain   bool   `json:"is_abstain"`
+	IsNo        bool   `json:"is_no"`
+}
+
+// GetVoteChoicesResult holds the stake version for the VoteChoices result
+// fetched and the AgendaVoteChoices.
+type GetVoteChoicesResult struct {
+	Version uint32             `json:"version"`
+	Choices []AgendaVoteChoice `json:"choices"`
+}
+
+// AgendaVoteChoice holds information about the current vote choice of an agenda
+type AgendaVoteChoice struct {
+	AgendaID          string `json:"agenda_id"`
+	AgendaDescription string `json:"agenda_description"`
+	ChoiceID          string `json:"choice_id"`
+	ChoiceDescription string `json:"choice_description"`
+}
+
+// ByStartTime implements sort.Interface based on the Agenda.StartTime field.
+type ByStartTime []*Agenda
+
+func (a ByStartTime) Len() int { return len(a) }
+func (a ByStartTime) Less(i, j int) bool {
+	return a[i].StartTime > a[j].StartTime
+}
+func (a ByStartTime) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
```

Comment:
A previous review recommended replacing this with a slice sorter that can be used e.g.:
```
sort.Slice(agendas, func(i, j int) bool {
	return a[i].StartTime > a[j].StartTime
})
```

Reply:
oh, didn't come across that review